app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectService.kt

```diff
@@ -271,7 +281,33 @@ class WalletConnectService : Service(), SignClient.WalletDelegate, CoreClient.Co
         EventBus.getDefault().post(ConnectionState(false))
     }
 
+
+    val expectedNetwork: String = BuildConfig.EXPORT_CHAIN
+    val expectedNamespaceKey = "ccd"
+    val expectedChain = "$expectedNamespaceKey:$expectedNetwork"
+    val signAndSendTransactionMethod = "sign_and_send_transaction"
+    val signMessageMethod = "sign_message"
+    val supportedChains = listOf(expectedChain)
+    val supportedEvents = listOf("accounts_changed", "chain_changed")
+    val supportedMethods = listOf(signAndSendTransactionMethod, signMessageMethod)
     override fun onSessionProposal(sessionProposal: Sign.Model.SessionProposal) {
+        val ccdNamespace = sessionProposal.requiredNamespaces[expectedNamespaceKey]
+        if (ccdNamespace?.chains != supportedChains) {
+            EventBus.getDefault()
+                .post(RejectError("Expected namespaces:${supportedChains} but got ${sessionProposal.requiredNamespaces.keys}"))
```

Comment:
```suggestion
                .post(RejectError("Expected namespaces:${supportedChains} but got ${ccdNamespace?.chains}"))
```
using keys I’m not getting information about what namespace it asked for.